Content.Shared/Chemistry/ReagentUnit.cs

```diff
@@ -0,0 +1,184 @@
+﻿using Robust.Shared.Interfaces.Serialization;
+using Robust.Shared.Serialization;
+using System;
+using System.Linq;
+
+namespace Content.Shared.Chemistry
+{
+    [Serializable]
+    public struct ReagentUnit : ISelfSerialize
+    {
+        private int _value;
+        private static readonly int Shift = 2;
+
+        public static ReagentUnit MaxValue => new ReagentUnit(int.MaxValue);
+
+        private decimal ShiftDown()
+        {
+            return _value / (decimal)Math.Pow(10, Shift);
+        }
+
+        private ReagentUnit(int value)
+        {
+            _value = value;
+        }
+
+        public static ReagentUnit New(int value)
+        {
+            return new ReagentUnit(value * (int) Math.Pow(10, Shift));
+        }
+
+        public static ReagentUnit New(decimal value)
+        {
+            return new ReagentUnit((int) Math.Round(value * (decimal) Math.Pow(10, Shift)));
+        }
+
+        public static ReagentUnit New(float value)
+        {
+            return new ReagentUnit(FromFloat(value));
+        }
+
+        private static int FromFloat(float value)
+        {
+            return (int) Math.Round(value * (float) Math.Pow(10, Shift));
+        }
+
+        public static ReagentUnit New(double value)
+        {
+            return new ReagentUnit((int) Math.Round(value * Math.Pow(10, Shift)));
+        }
+
+        public static ReagentUnit New(string value)
+        {
+            return New(FloatFromString(value));
+        }
+
+        private static float FloatFromString(string value)
```

Comment:
You're gonna need a way to pass through the `CultureInfo` or such too here, because right now this uses current-culture parsing meaning that `Deserialize` doesn't work correctly when your culture is, say, french.

Reply:
1. Added
2. Not sure what you mean by this but I added some unit tests for some nasty fractional cases and they all seem to be handled well.

Reply:
I'm just passing invariant culture to all the string things./